section1/reflection.md

```diff
@@ -1,19 +1,57 @@
 
+// Two backslashes are used for a single line of a comment.
+
+/* If you want to comment out multiple lines of code, you can use a backslash followed by an asterisk and then wrap the code with an asterisk followed by a backslash. */
+* The are multiple reasons for code comments. We try to write code as clearly as possible, but sometimes it requires a little extra explanation. Sometimes we use comments to organize code into sections. Sometimes we want to comment out a piece of code because we're testing another function or something isn't working.
+
 5. In your own words, what is a variable? How would you explain it to a 5 year old?
+* A variable is like a box that allows you to put something in it. We call that something an object. Once you know what or decide what that object is you can name it. Once you've give that object a name you can declare it or give it a value. For example, I can make a variable an animal by saying `var animal`, then I have to decide what that animal is. The animal can be anything I decide. Let's make the animal a panda by saying `var animal = "panda"`.
 
 6. Think of a site or app you use frequently. What are three variables that are probably used? Which data type would each of those variables probably hold?
+* I frequently visit the New York Times, both in app form and the website. Three variables and the data types each probably uses are:
+1. `var weather = 65`. This is would be a **Number** data type - the weather and date are always up to date.
+2. `var name = ""` if empty = false, `var name = "Dinne"` else is true. This would be a **Boolean** - the login would take you through a true/false set of __else...if__ statements. There is a quiz feature for example, when you answer the question, it prompts you through the next question, by going through true/false.
+3. `var message = "Congratulations"`. You get a "congratulations" when you finish the quiz, this is a **String**
 
 7. In your own words, explain what concatenation is.
+* Concatenation is adding two or more strings together into one string.
 
 8. Think of a site or app you use frequently. Where do you think the developers used concatenation?
+* I often peruse a clothing site called Lulus. One of the places the developers probably used concatenation is in the greeting. When you've signed into the site an alert pops up "Hi Dinne!" I imagine the code goes something like this:
+
+ ```javascript
+ var firstName = "Dinne";
+ var lastName = "Kopelevich";
+
+ console.log("Hi " + firstName + "!")
+ ```
+
+ Or they use interpolation instead.
+
 
 9. What questions do you still have about the work we've done so far? (not having a question is not an option)
+* Why exactly do we use fork before cloning and after making changes, doing a pull requests, if a team is working on the same project? 
+* Why do we initially have to have to assign a value to a boolean data type?
```

Comment:
I believe you could assign it as nil and then assign it as true or false later based on a condition. I work mostly in Python and Java and I see that from time to time.